### DISCUSSION TITLE
What is the best way to distribute a solidjs component?

### DISCUSSION THREAD

**Discussion by ciaoben** [solidjs/solid#839] [Q&A] [answered]
I am falling in love with solid. It is really amazing, thank you for this library!

I need to create some components to be used in various frameworks and I am exploring the idea to use solid. I saw `solid-element` but I am a little scared of webcomponents' SSR problems with Shadow DOM and performances.
I was wondering if it is the best way to distribute a quite large set of components usable by other frameworks or I am missing something? 

Is solid a good choice for this use case?

**Comment by ryansolid** [comment, marked as answer]
Usable by other frameworks generally there is never a good solution. Usually web components is the best answer. Like React components and Svelte components aren't really usable by other framework. You could author things in a widget like way that independently mount, but that is very DOM specific.

The truth of the matter is there aren't any really general solutions for cross framework components when you are considering SSR. Since they all render differently and hydrate differently. Even attempts on the webcomponent side still tie you to a specific framework like Lit. In general framework agnostic components that aren't based on static html(like bootstrap style) haven't really ever been accomplished. As soon as you bring in declarative library they are all incompatible with each other or require a bunch of specific wiring at component boundaries. Which as you imagine in a case like this would be everywhere.

**Reply by ciaoben** [reply]
Thanks @ryansolid for the reply. You have indeed confirmed my fears. I hope in the future custom-element will be more powerful. 
Right now it seems that the only solution is to used solid to create custom elements and then write the wrapper around them specific for each framework a la https://shoelace.style/.

I am interested in your opinion, do you find a viable idea?

**Reply by milahu** [reply]
related:

- micro-frontends (overhead from web workers)
- mitosis https://github.com/BuilderIO/mitosis and "reverse mitosis" https://github.com/BuilderIO/mitosis/issues/149
- converters https://github.com/milahu/awesome-component-converters
